### PR TITLE
[DomCrawler] exclude fields inside "template" tags

### PR DESCRIPTION
**Pull request by Gorjunov**
| Q             | A
| ------------- | ---
| Branch?       | 2.8
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #27285 
| License       | MIT
| Doc PR        | 

Exclude fields values/files if fields are inside template tag. I think better to exclude values only instead of excluding fields at all (described in ticket #27285)

### CODE REVIEW

File: src/Symfony/Component/DomCrawler/Tests/FormTest.php [hunk 1]
```diff
@@ -847,7 +853,7 @@ protected function getFormFieldMock($name, $value = null)
     protected function createForm($form, $method = null, $currentUri = null)
     {
         $dom = new \DOMDocument();
-        $dom->loadHTML('<html>'.$form.'</html>');
```
Comment (jakzal) [inline src/Symfony/Component/DomCrawler/Tests/FormTest.php]: Why did you make this change?

**Comment by jakzal** [comment]
>  I think better to exclude values only instead of excluding fields at all 

Could you elaborate on why it's better?

File: src/Symfony/Component/DomCrawler/Tests/FormTest.php (hunk 1 above)
Comment (Gorjunov) [reply src/Symfony/Component/DomCrawler/Tests/FormTest.php]: Now I'm testing form with HTML5 element. Without `@` `loadHTML` method will return warning type of `Tag template invalid in Entity...`

**Comment by Gorjunov** [comment]
In this case it represents browser's behavior - element is accessible (e.g. by JS), but server will not get values on form submit. (if I understand right that `Form:: getValues()` is same as form submit in browser)

**Comment by nicolas-grekas** [comment]
The way this is currently implemented could be understood as a new feature because it's adding a new method. If we can do without the new method, that'd be better to qualify as bugfix.

> better to exclude values only instead of excluding fields at all

Shouldn't the `$fields` be *not* populated with tags inside template instead?

**Review by nicolas-grekas — APPROVED** [review]
(implementation updated to filter fields inside `<template>` using xpath)

**Comment by stof** [comment]
Tests should also be updated to test that the Form does not contain the corresponding field at all (`has` returns false for this field name)

**Comment by nicolas-grekas** [comment]
@stof test added

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Gorjunov.

**Pull request merged**